File: src/Java/701-800/746.MinCostClimbingStairs.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;
import java.util.HashSet;

class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int length = cost.length;
        if (length == 1)
            return cost[0];
        int i = 2;
        for (; i < length; i++) {
            cost[i] = cost[i] + Math.min(cost[i - 1], cost[i - 2]);
        }   	
        return Math.min(cost[i - 1], cost[i - 2]);
    }
}

File: src/Python/701-800/746.MinCostClimbingStairs.py

class Solution:
    def minCostClimbingStairs(self, cost):
        """
        :type cost: List[int]
        :rtype: int
        """
        p_2, p_1 = cost[0], cost[1] 
        for c in cost[2:]: 
            p_2, p_1 = p_1, min(p_1 + c, p_2 + c) 
        return min(p_2, p_1)

if __name__ == '__main__':
    solution = Solution()
    print(solution.minCostClimbingStairs([10, 15, 20]))
    print(solution.minCostClimbingStairs([1, 100, 1, 1, 1, 100, 1, 1, 100, 1]))
    print(solution.minCostClimbingStairs([10, 15, 20, 10]))
    print(solution.minCostClimbingStairs([0, 1, 2, 2]))
else:
    pass

